keras_cv/models/backbone.py

```diff
@@ -0,0 +1,197 @@
+        """Instantiate {{model_name}} model from preset architecture and weights.
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+            load_weights: Whether to load pre-trained weights into model.
+                Defaults to `True`.
+        Examples:
+        ```python
+        # Load architecture and weights from preset
+        model = keras_cv.models.{{model_name}}.from_preset(
+            "{{example_preset_name}}"
+        )
+        # Load randomly initialized model from preset architecture
+        model = keras_cv.models.{{model_name}}.from_preset(
+            "{{example_preset_name}}",
+            load_weights=False
+        )
+        ```
+        """
+
+        if not cls.presets:
+            raise NotImplementedError(
+                "No presets have been created for this class."
+            )
+
+        if preset not in cls.presets:
+            raise ValueError(
+                "`preset` must be one of "
+                f"""{", ".join(cls.presets)}. Received: {preset}."""
+            )
+        metadata = cls.presets[preset]
+
+        if load_weights and "weights_url" not in metadata:
+            raise ValueError(
+                f"""Pretrained weights not available for preset "{preset}". """
+                "Set `load_weights=False` to use this preset or choose one of "
+                "the following presets with weights:"
+                f""" "{'", "'.join(cls.presets_with_weights)}"."""
+            )
+
+        config = metadata["config"]
+        model = cls.from_config({**config, **kwargs})
+
+        if not load_weights:
+            return model
+
+        weights = keras.utils.get_file(
+            "model.h5",
+            metadata["weights_url"],
+            cache_subdir=os.path.join("models", preset),
+            file_hash=metadata["weights_hash"],
+        )
+
+        model.load_weights(weights)
+        return model
+
+    def __init_subclass__(cls, **kwargs):
+        # Use __init_subclass__ to setup a correct docstring for from_preset.
+        super().__init_subclass__(**kwargs)
+
+        # If the subclass does not define from_preset, assign a wrapper so that
+        # each class can have an distinct docstring.
+        if "from_preset" not in cls.__dict__:
+
+            def from_preset(calling_cls, *args, **kwargs):
+                return super(cls, calling_cls).from_preset(*args, **kwargs)
+
+            cls.from_preset = classmethod(from_preset)
+
+        # Format and assign the docstring unless the subclass has overridden it.
+        if cls.from_preset.__doc__ is None:
+            cls.from_preset.__func__.__doc__ = Backbone.from_preset.__doc__
+            format_docstring(
+                model_name=cls.__name__,
+                example_preset_name=next(iter(cls.presets_with_weights), ""),
+                preset_names='", "'.join(cls.presets),
+            )(cls.from_preset.__func__)
+
+    @property
+    def backbone_level_outputs(self):
```

Comment:
This is unclear to me

Reply:
This is some logic from KerasNLP that copies the templated docstring here to the subclass. It's not something we expect users to interact with.